utilities/flink/flink_compaction_filter.cc

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#include <algorithm>
+#include <iostream>
+#include "utilities/flink/flink_compaction_filter.h"
+
+namespace rocksdb {
+namespace flink {
+
+int64_t DeserializeTimestamp(const char *src, std::size_t offset) {
+  uint64_t result = 0;
```

Comment:
I wonder if this is he appropriate type, because longs in java are signed and negative timestamps can exists in event time in flink. I think the test should cover negative timestamps as well.

Reply:
I am not sure the bit operations will work correctly with signed type, at the end the function returns negative number. I will also extend the test to generate random negative numbers.